### PR TITLE
General: Move load related functions into pipeline

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Loading workflow specific functions were moved to `openpype.pipeline.load` and their related to other places.

## Description
Implemented functions to filter containers into few categories to be able get outdated containers or check for them easier. Added helper client function to find out if version id is latest version.

Functions marked as deprecated in `openpype.lib` are `get_latest_version`, `is_latest` and `any_outdated`.

## Testing notes:
Submission to deadline should properly detect last version.
Check for outdated containers should work in hosts
- [ ] Aftereffects
- [x] Harmony - loading should also mark outdated containers with red and latest with green color
- [x] Houdini
- [x] Maya
- [x] Photoshop

### CODE REVIEW

File: openpype/client/entities.py [hunk 1]
```diff
@@ -557,6 +557,42 @@ def get_version_by_name(project_name, version, subset_id, fields=None):
...
+        bool: True if is latest version from subset else False.
+    """
+
+    version_id = _convert_id(version_id)
+    if not version_id:
+        return False
+    version_doc = get_version_by_id(
+        project_name, version_id, fields=["_id", "type", "parent"]
+    )
+    # What to de when version is not found?
```
Comment (antirotor) [inline openpype/client/entities.py:583]:
```suggestion
    # What to when version is not found?
```

just a simple typo

**Review by antirotor — COMMENTED** [review]
Haven't tested it yet but the code looks solid.

**Comment by 64qam** [comment]
Validators in AE and PS are ignored

**Comment by iLLiCiTiT** [comment]
Not sure if it's related to this PR but there is error happening in Harmony:
```
Traceback (most recent call last):
File "C:\Users\bumpybox\AppData\Local\pypeclub\openpype\openpype-v3.12.2-nightly.2\openpype\tools\sceneinventory\view.py", line 246, in 

lambda: _on_update_to_latest(items)
File "C:\Users\bumpybox\AppData\Local\pypeclub\openpype\openpype-v3.12.2-nightly.2\openpype\tools\sceneinventory\view.py", line 230, in _on_update_to_latest

update_container(item, -1)
File "C:\Users\bumpybox\AppData\Local\pypeclub\openpype\openpype-v3.12.2-nightly.2\openpype\pipeline\load\utils.py", line 439, in update_container

return loader.update(container, new_representation)
File "C:\Users\bumpybox\AppData\Local\pypeclub\openpype\openpype-v3.12.2-nightly.2\openpype\hosts\harmony\plugins\load\load_imagesequence.py", line 83, in update

node = container.get("nodes").pop()
Index ERROR 

pop from empty list
```

**Review by kalisp — APPROVED** [review]
Issues in Harmony were solved.
AE was missing in `validate_containers`, added and tested.

**Pull request merged**